src/main/java/frc/robot/subsystems/VisionSystem.java

```diff
@@ -0,0 +1,40 @@
+package frc.robot.subsystems;
+
+import edu.wpi.first.apriltag.AprilTagFieldLayout;
+import edu.wpi.first.apriltag.AprilTagFields;
+import edu.wpi.first.wpilibj.DriverStation;
+import edu.wpi.first.wpilibj2.command.SubsystemBase;
+import frc.robot.Limelight;
+import frc.robot.LimelightHelpers;
+
+import java.util.Optional;
+
+public class VisionSystem extends SubsystemBase {
+
+    private static final String LL_NAME_BANANA = "limelight-banana";
+    private static final String LL_NANE_APPLE = "limelight-apple";
+    private boolean redAlliance;
+
+    private final Limelight limelightBack;
+    private final Limelight limelightFront;
+    private final AprilTagFieldLayout layout;
+    public VisionSystem(){
+        limelightBack = new Limelight(LL_NAME_BANANA);
+        limelightFront = new Limelight(LL_NANE_APPLE);
+        layout = AprilTagFieldLayout.loadField(AprilTagFields.kDefaultField);
+        if(DriverStation.getAlliance().isPresent())
```

Comment:
loading this in the ctor won't work well, as this changes on connection to the FMS/DS, which occurs after the robot boots. You can query this on demand instead.

Reply:
this code is in the constructor of the class. The instance is initialized in `robotInit`, so the code runs when the robot code starts.

This is problematic. You see, this information comes directly from the DriverStation (or FMS in competitions). This means that this information is not available when the robot code starts, but rather only later on. Additionally, when testing at home, we can always change this value, so it is not constant.

As such, you should check the alliance on demand. That is, check it in the method call you need it for and not in the constructor. It is not a constant, and is not available on start.